lib/topological_inventory/orchestrator/worker.rb

```diff
@@ -116,9 +97,67 @@ def pg_connection_args
         }.freeze
       end
 
-      def clear_caches
-        @sources = nil
-        @collector_deployments = nil
+
+      ### Orchestrator Stuff
+      def collector_definitions
+        @collector_definitions ||= begin
+          require 'yaml'
+          YAML.load_file(@collector_definitions_file)
+        end
+      end
+
+      def object_manager
+        @object_manager ||= ObjectManager.new
+      end
+
+
+      ### Openshift stuff
+      def collectors_from_openshift
+        object_manager.get_deployment_configs("topological-inventory/collector=true").collect { |i| i.metadata.labels["topological-inventory/collector_digest"] }
+      end
+
+      def create_openshift_objects_for_source(digest, source)
+        puts "Creating objects for source #{source["source_id"]} with digest #{digest}"
+        object_manager.create_secret(collector_deployment_secret_name_for_source(source), {"openshift-token" => source["token"]})
```

Comment:
@agrare Any thoughts?

Reply:
I don't really like `id` because it could be confused with the `id` of the record.  Maybe `public` and `private`?

Reply:
Hey sorry didn't see this, providers fall into three groups:
1. Typical username / password (most infra providers)
2. public / private keys (aws/azure)
3. Just a private key (openshift/kubernetes)

Public/private would cover all these

Reply:
After some more discussions with @carbonin and @Fryguy we decided `private` and `public` were probably not great names since they're both ruby keywords.  However `username` and `password` are the very popular and clearly understood, so we'll just stick with that for now.

Reply:
:+1: awesome